store/displayPreferences.ts

```diff
@@ -1,116 +1,104 @@
-import Vue from 'vue';
 import { DisplayPreferencesDto } from '@jellyfin/client-axios';
-import { MutationTree, ActionTree, GetterTree } from 'vuex';
-import nuxtConfig from '~/nuxt.config';
+import { MutationTree, ActionTree } from 'vuex';
+import { merge, toString as toStr } from 'lodash';
+import { boolean } from 'boolean';
 
-const stringToBoolean = (value: string): boolean => value === 'True';
-const booleanToString = (value: boolean): string => (value ? 'True' : 'False');
-
-export interface DisplayPreferencesState extends DisplayPreferencesDto {
-  CustomPrefs: {
-    [key: string]: string;
-  };
+/**
+ * TODO: Generate this from the nuxtConfig file
+ */
+export enum SupportedLocales {
+  auto = 'auto',
+  cs = 'cs',
+  de = 'de',
+  enUs = 'enUs',
+  es = 'es',
+  fr = 'fr',
+  nb = 'nb',
+  nl = 'nl',
+  pl = 'pl',
+  ro = 'ro',
+  sk = 'sk',
+  sl = 'sl',
+  sv = 'sv',
+  ta = 'ta',
+  tr = 'tr',
+  vi = 'vi',
+  zhCn = 'zhCn'
 }
 
-const defaultCustomPrefs: {
-  [key: string]: string;
-} = {
-  darkMode: booleanToString(
-    nuxtConfig.vuetify?.theme?.default === 'dark' || false
-  ),
-  locale: 'auto'
-};
+/**
+ * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+export interface CustomPreferences {
+  darkMode: boolean;
+  locale: SupportedLocales;
+}
+/**
+ * DisplayPreferencesDto but with our custom typings for CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientPreferences
+  extends Omit<DisplayPreferencesDto, 'CustomPrefs'> {
+  CustomPrefs?: CustomPreferences;
+}
+/**
+ * Typings for Vuex's store state
```

Comment:
We're using those for normal comments now as they provide automatic wrapping and newlines with the VSCode toolchain we have. We're trying to avoid the normal comments except for eslint exceptions.

That makes it easy to check also at a go which comments are useful and which ones are related to eslint stuff.

Reply:
@Codex- Oh I didn't understood what you mean at first.

I placed them here as they might lead to confusion: the store state isn't the same as what the CustomPrefs of the server includes. The syncing boolean for example isn't synced with the server and it belongs only to the store, that's why I think it could lead to some confusions.

Ofc the comment alone means nothing, if you see it as a block of all the commented typings I think it makes sense. 

Thoughts @Camc314 @ThibaultNocchi ?

Reply:
I agree with Codex, we're already particularly used to the syntax xxxxState for our state classes in the store, and this file isn't a weird mess of stuff yet so I think it's easy to figure that this isn't what's sent to the server. Though if you wish you could add a comment for the `syncing` attribute stating like "extraneous attribute to watch for syncing"